docs/virtual_machines/disks_and_volumes.md

```diff
@@ -740,91 +806,214 @@ the created kubernetes `Secret`.
+        bootcmd:
+          # mount the Secret
+          - "mkdir /mnt/app-secret"
+          - "mount /dev/$(lsblk --nodeps -no name,serial | grep D23YZ9W6WA5DJ487 | cut -f1 -d' ') /mnt/app-secret"
+    name: cloudinitdisk
+  - secret:
+      secretName: app-secret
+    name: app-secret-disk
+status: {}
+```
+
+#### Using filesystem
+
+By using filesystem, `secrets` are shared through `virtiofs`. In contrast with using disk for sharing `secrets`,
+`filesystem` allows you to dynamically propagate changes on `secrets` to VMIs (i.e., the VM does not need to be rebooted).
+
+> **Note:** The **ExperimentalVirtiofsSupport** feature gate
+> [must be enabled](../operations/activating_feature_gates.md#how-to-activate-a-feature-gate)
+> to share directories with VMs. This feature is available starting with KubeVirt v0.36.0.
+
+> **Note:** Currently, VMIs can not be live migrated since `virtiofs` does not support live migration.
+ 
+To share a given `secret`, the following VM definition could be used:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-fedora
+  name: vmi-fedora
+spec:
+  domain:
+    devices:
+      filesystems:
+        - name: app-secret-fs
+          virtiofs: {}
+      disks:
+        - disk:
+            bus: virtio
+          name: containerdisk
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  terminationGracePeriodSeconds: 0
+  volumes:
+    - name: containerdisk
+      containerDisk:
+        image: quay.io/containerdisks/fedora:latest
+    - cloudInitNoCloud:
+        userData: |-
+          #cloud-config
+          chpasswd:
+            expire: false
+          password: fedora
+          user: fedora
+          bootcmd:
+            # mount the Secret
+            - "sudo mkdir /mnt/app-secret"
+            - "sudo mount -t virtiofs app-secret-fs /mnt/app-secret"
+      name: cloudinitdisk
+    - secret:
+        secretName: app-secret
+      name: app-secret-fs
+```
 
 ### serviceAccount
 
 A `serviceAccount` volume references a Kubernetes
 [`ServiceAccount`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/).
-A new `iso` disk will be allocated with the content of the service
+
+#### Using disk
+By using disk, a new `iso` disk will be allocated with the content of the service
 account (`namespace`, `token` and `ca.crt`), which needs to be mounted
 in the VM. For automatic mounting, see the `configMap` and `secret`
 examples above.
 
+> **Note:** Currently, ServiceAccount update propagation is not supported. If a
+> ServiceAccount is updated, only a pod will be aware of changes, not running
+> VMIs.
+
 Example:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-fedora
+  name: vmi-fedora
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
```

Comment:
In this example you've used `virtio` as the disk bus so adding a disk is required without preferences but assuming the bus isn't important you can again drop this from the example to illustrate the default behaviour.

Reply:
I've dropped the `virtio` bus from the example. Do you think that I should drop the `bus`  from the other examples as well?